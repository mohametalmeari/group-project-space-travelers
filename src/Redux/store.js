import { configureStore } from '@reduxjs/toolkit';
import missionReducer from './features/missions/missionSlice';

const store = configureStore({
  reducer: {
    mission: missionReducer,
  },
});

export default store;
