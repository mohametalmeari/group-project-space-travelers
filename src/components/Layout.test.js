import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import store from '../Redux/store';
import Layout from './Layout';

describe('Layout Component', () => {
  test('Test render of Layout', () => {
    const { container } = render(
      <MemoryRouter>
        <Provider store={store}>
          <Layout />
        </Provider>
      </MemoryRouter>,
    );
    expect(container).toMatchSnapshot();
  });
});
