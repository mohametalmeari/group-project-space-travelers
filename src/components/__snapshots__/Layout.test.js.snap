// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout Component Test render of Layout 1`] = `
<div>
  <div
    class="wrapper"
  >
    <nav>
      <h1>
        <a
          aria-current="page"
          class="active"
          href="/"
        >
          Space Travelers Hub
        </a>
      </h1>
      <ul
        class="menu"
      >
        <li
          class="page-link"
        >
          <a
            aria-current="page"
            class="active"
            href="/"
          >
            Rockets
          </a>
        </li>
        <li
          class="page-link"
        >
          <a
            class=""
            href="/missions"
          >
            Missions
          </a>
        </li>
        <li
          class="page-link"
        >
          <a
            class=""
            href="/my-profile"
          >
            My Profile
          </a>
        </li>
      </ul>
    </nav>
  </div>
</div>
`;

exports[`Layout Component test render of Layout 1`] = `
<div>
  <div
    class="wrapper"
  >
    <nav>
      <h1>
        <a
          aria-current="page"
          class="active"
          href="/"
        >
          Space Travelers Hub
        </a>
      </h1>
      <ul
        class="menu"
      >
        <li
          class="page-link"
        >
          <a
            aria-current="page"
            class="active"
            href="/"
          >
            Rockets
          </a>
        </li>
        <li
          class="page-link"
        >
          <a
            class=""
            href="/missions"
          >
            Missions
          </a>
        </li>
        <li
          class="page-link"
        >
          <a
            class=""
            href="/my-profile"
          >
            My Profile
          </a>
        </li>
      </ul>
    </nav>
  </div>
</div>
`;
